### DISCUSSION TITLE
Does Keycloak support Backchannel logout for Amazon Cognito?

### DISCUSSION THREAD

**Discussion by shawnhankim** [keycloak/keycloak#30353] [Q&A]
Hello,

I am using Keycloak and several identity providers using OIDC. But the Keycloak's backchannel logout doesn't work with Amazon Cognito unlike other IdPs such as Azure Entera and Okta.

I am wondering if Keycloak doesn't support backchannel logout for Amazon Cognito, or if I miss anything. FYI: The general logout works without clearing sessions by adding the query parameters of `client_id` and `logout_uri` which are required for Cognito logout.

Looking forward to hearing from you.

**Comment by riazaham** [comment]
Hi @shawnhankim,

How did you get the general logout to work? I am not able to do a full logout as I get a 400 bad request error after Cognito redirects back to KeyCloak. I have also included the `client_id` and `logout_uri` as params.

It seems like Cognito's params mismatch with KeyCloak's expected params.

Looking forward to hearing from you
